### DISCUSSION TITLE
master vs development branch

### DISCUSSION THREAD

**Discussion by jankapunkt** [node-oauth/node-oauth2-server#233] [General]
@jorenvandeweyer I found that when I change the base branch to `development` then GitHub makes not only all PR incoming to `development` but also makes it the default page and also our docs target it.

The problem is, that `development` might not represent a stable state of the project, which is IMO very bad as default page when viewing the repo.

I will revert it for now to `master` but we should think about a usable and practical git strategy that allows us easily to manage multiple major versions but keep it simple at the core.

What do you think?

**Comment by jorenvandeweyer** [comment]
Hm, in that case we should keep the master branch the main branch.

From the moment we mark V5 as stable the inconvenience will be a lot less since we'll be able to merge development more often into the master branch.

**Reply by jankapunkt** [reply]
Yes, exactly my thoughts

**Comment by jorenvandeweyer** [comment]
@jankapunkt 

Just to make sure I'm targeting my PR's correctly.

Security updates should be merged:
- into the `master` branch
- into the `development` branch

V5 features and rewrites should be merged:
- into the `development` branch

**Reply by jankapunkt** [reply]
> Security updates should be merged:
>
> - into the master branch
> - into the development branch

For now this is the way to go, since v4 and v5 are too different from each other

> V5 features and rewrites should be merged

I would propose to merge them directly into `release-5.0.0`. Once we have the final 5.0.0 out we can merge all back into development and master.

Does this make sense?

**Reply by jorenvandeweyer** [reply]
To be honest I don't know what is best. I was thinking about it because I see some merge conflicts between `release-5.0.0` & `development`

The easiest way to prevent merge conflicts is to push everything to `development` and merge that branch regularly into `release-5.0.0`.

I'd like to start using 5.x in some projects of mine so we can start testing if it's stable.

**Reply by jankapunkt** [reply]
> To be honest I don't know what is best. I was thinking about it because I see some merge conflicts between release-5.0.0 & development

I already arranged some time the next days to make a clean merge of `development` into `release-5.0.0` to no worries on that, I'm used to resolve mc at work ^_^

> The easiest way to prevent merge conflicts is to push everything to development and merge that branch regularly into release-5.0.0.

Okay then let's target `development` and I will work on merging it into release-5.0.0

> I'd like to start using 5.x in some projects of mine so we can start testing if it's stable.

I think it's on the verge to become stable, I already fixed some issues, was mostly some missing `await` calls with help of @shrihari-prakash (he actually fixed them)

**Reply by jankapunkt** [reply]
I merged `development` into `5.0.0` and resolved lots of conflicts but the deep integration tests paid off already! Threw a couple of errors so I found some things that were overridden by merge.

**Reply by jorenvandeweyer** [reply]
@jankapunkt shouldn't we also merge 5.0.0 into development, because it's a lot of commits ahead?

**Reply by jankapunkt** [reply]
@jorenvandeweyer you are right, I also commented this on the release-5.0.0 pr https://github.com/node-oauth/node-oauth2-server/pull/194#issuecomment-1696868076
